pyvista/core/filters/poly_data.py

```diff
@@ -2755,6 +2756,107 @@ def extrude_rotate(
+        """Extrude polygonal data trimmed by a surface.
+
+        This generates polygonal data on output.
+
+        The input dataset is swept along a specified direction forming a
+        "skirt" from the boundary edges 2D primitives (i.e., edges used
+        by only one polygon); and/or from vertices and lines. The extent
+        of the sweeping is defined where the sweep intersects a
+        user-specified surface.
+
+        Parameters
+        ----------
+        direction : numpy.ndarray or sequence
+            Direction vector to extrude.
+
+        trim_surface: pyvista.PolyData
+            Surface which trims the surface.
+
+        extrusion : str or int, optional
+            Control the strategy of extrusion. One of the following:
+
+            * ``"boundary_edges"`` or ``0``
+            * ``"all_edges"`` or ``1``
+            The default is "boundary_edges".
+
+        capping : str or int, optional
+            Control the strategy of capping. One of the following:
+
+            * ``"intersection"`` or ``0``
+            * ``"minimum_distance"`` or ``1``
+            * ``"maximum_distance"`` or ``2``
+            * ``"average_distance"`` or ``3``
+            The default is "intersection".
+
+        inplace : bool, optional
+            Overwrites the original mesh in-place.
+
+        progress_bar : bool, optional
+            Display a progress bar to indicate progress.
+
+        Returns
+        -------
+        pyvista.PolyData
+
+        Examples
+        --------
+        Extrude a beam from surface.
+
+        >>> import pyvista
+        >>> import numpy as np
+        >>> direction = np.array([0, 0, 2])
+        >>> plane = pyvista.Plane()
+        >>> disc = pyvista.Disc(center=(0, 0, 0))
+        >>> extruded = disc.extrude_trim(direction, plane)
+        >>> extruded.plot(color="tan")
+
+        """
+        if not isinstance(direction, (np.ndarray, collections.abc.Sequence)) or len(direction) != 3:
+            raise TypeError('Vector must be a length three vector')
+        extrusions = {"boundary_edges": 0, "all_edges": 1}
+        if isinstance(extrusion, str):
+            if extrusion not in extrusions:
+                raise ValueError(f'Invalid strategy of extrusion "{extrusion}".')
+            extrusion = extrusions[extrusion]
+        elif isinstance(extrusion, int) and extrusion not in range(2):
+            raise ValueError(f'Invalid strategy of extrusion index "{extrusion}".')
+        elif not isinstance(extrusion, int):
+            raise ValueError(f'Invalid strategy of extrusion index type "{type(extrusion).__name__}".')
+
+        cappings = {"intersection": 0, "minimum_distance": 1, "maximum_distance": 2, "average_distance": 3}
+        if isinstance(capping, str):
+            if capping not in cappings:
+                raise ValueError(f'Invalid strategy of capping "{capping}".')
+            capping = cappings[capping]
+        elif isinstance(capping, int) and capping in range(3):
```

Comment:
Should be `capping not in range(3)`?

Reply:
You are right. Thanks :)
```suggestion
        elif isinstance(capping, int) and capping not in range(4):
```